pandas/tests/io/excel/test_style.py

```diff
@@ -333,3 +336,25 @@ def test_styler_to_s3(s3_public_bucket, s3so):
             f"s3://{mock_bucket_name}/{target_file}", index_col=0, storage_options=s3so
         )
         tm.assert_frame_equal(result, df)
+
+
+@pytest.mark.parametrize("merge_cells", [True, False, "columns"])
+def test_format_hierarchical_rows_periodindex(merge_cells):
+    # GH#60099
+    df = DataFrame(
+        {"A": [1, 2]},
+        index=MultiIndex.from_arrays(
+            [
+                period_range(start="2006-10-06", end="2006-10-07", freq="D"),
+                ["X", "Y"],
+            ],
+            names=["date", "category"],
+        ),
+    )
+    formatter = ExcelFormatter(df, merge_cells=merge_cells)
+    formatted_cells = list(formatter._format_hierarchical_rows())
```

Comment:
```suggestion
    formatted_cells = formatter._format_hierarchical_rows()
```
Seems not necessary to convert this generator to a list? (to save memory)

Reply:
You are right! I'll change it soon.

Reply:
Thanks!